### ISSUE TITLE
Cache Operation

### ISSUE THREAD

**Issue by luc345** [apollographql/apollo-client-nextjs#147]
Environment:
@apollo/experimental-nextjs-app-support: "0.5.2"
@apollo/client: "^3.8.0-rc.2"
next.js: "14.03"

I have an SSR query initialized as follows:
```
function makeClient() {
  const httpLink = new HttpLink({
    uri: "http://localhost:4000/",
    fetchOptions: { cache: "no-store" },
    credentials: "include",
  });
  return new NextSSRApolloClient({
    connectToDevTools: true,
    cache: new NextSSRInMemoryCache(),
    link:
      typeof window === "undefined"
        ? ApolloLink.from([
            new SSRMultipartLink({
              stripDefer: true,
            }),
            httpLink,
          ])
        : httpLink,
  });
}
```

Query:
```
const {
  data,
  loading,
  error,
} = useQuery(GET_DATA, {
  variables: { ... },
});
```

I have two questions..
1. When I set fetchOptions: { cache: "no-store" } for the query, I noticed in Apollo Client Developer Tools that the cache still contains the data from the previous query.
2. When I found that Apollo Client Developer Tools showed the cache for my query, I tried to re-query, but it didn't use the cache. I attempted to add fetchPolicy: "cache-and-network" to the GET_DATA query, but I still couldn't successfully retrieve data from the cache.

Please assist me in correctly utilizing Apollo Client's caching functionality. Much appreciated!

**Comment by phryneas** [comment]
`useQuery` will start a network request after the component rendered - and at that time, SSR will already be finished and only transport a loading state to the client. 
All we can do from this package's perspective is ensure that we prevent an unnecessary (as there's no way of transporting it) request on the server.

You probably want to use `useSuspenseQuery` instead, since that will actually pause rendering on the server until data is there and then transport the data over.

**Comment by jerelmiller** [comment]
Hey @luc345 👋 

Just to double check, you're trying to avoid storing data in the Apollo cache correct? You'll want to do this with a [`fetchPolicy`](https://www.apollographql.com/docs/react/data/queries/#supported-fetch-policies) of [`no-cache`](https://www.apollographql.com/docs/react/data/queries/#no-cache), which you'd add to your `useQuery` hook.

```ts
const { data } = useQuery(GET_DATA, { fetchPolicy: 'no-cache' })
```

`cache: 'no-store'` is a [Next.js](https://nextjs.org/docs/app/api-reference/functions/fetch#optionscache) cache API , which is probably why you're still seeing that data in the Apollo Cache (so many caches!)

**Comment by luc345** [comment]
@phryneas
I apologize for providing incorrect information yesterday. I realized that the fetchPolicy: "cache-and-network" I used not only prioritizes the cache but also sends requests to the server to update it, ensuring fast web page loading and accurate cache data. Yesterday, I was unaware of this behavior, so when I observed queries being made to the server during my checks, I mistakenly thought the cache wasn't being utilized.

Today, I tried using useSuspenseQuery and useQuery for caching, and the results so far indicate that the cache is working correctly.

Thank you for your help :)

@jerelmiller
I implemented the method you suggested and indeed, I no longer see the cache in the Apollo Client Devtools. 

Thank you for your help :)
